packages/ref-core/src/cmip_ref_core/metrics.py

```diff
@@ -345,3 +349,61 @@ def run(self, definition: MetricExecutionDefinition) -> MetricResult:
         MetricResult
             The result of running the metric.
         """
+
+
+class CommandLineMetric(Metric):
+    """
+    Metric that can be run from the command line.
+    """
+
+    provider: Any  # TODO: CommandLineMetricsProvider causes circular import
```

Comment:
Since it is just for typing this can be worked around by adding the following above

```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from cmip_ref_core.providers import CommandLineMetricsProvider
```

I'm not 100% sure if the quotes are needed

```suggestion
    provider: "CommandLineMetricsProvider"
```

Reply:
Thanks! I couldn't get that to work before because then the interpreter complained that CommandLineMetricsProvider was not defined. Apparently, you need to either add quotes or use `from __future__ import annotations`.